crates/ty_ide/src/inlay_hints.rs

```diff
@@ -104,6 +123,10 @@ impl SourceOrderVisitor<'_> for InlayHintVisitor<'_> {
 
         match stmt {
             Stmt::Assign(assign) => {
+                if !self.settings.variable_types {
+                    return;
+                }
```

Comment:
It seems unnecessary to traverse the entire tree only to bail out at the assignment level. Should we add a `InlayHintSettings::any_enabled` and bail out in the inlay hint request if it returns false to avoid doing all this unnecessary work?

Reply:
Actually, I don't think we should add that today because there will be (in the future) multiple positions which we do not provide a way to disable. I don't think the server should provide configuring every position of inlay hints. So, I'm going to avoid adding this today.

Reply:
Can you give an example for inlays that the user can't disable and we'd always provide?